### DISCUSSION TITLE
Visual Studio vs Maui Nuget vs Workload Version

### DISCUSSION THREAD

**Discussion by epsmae** [dotnet/maui#9063] [Q&A]
In our development Team we would like to keep the tooling in sync and therefore I would like to better understand which versions are important for what.

I have a set of questions, hopefully someone is so kind to give me some more insights, thanks in advance

What is the role of the visual studio? Only ui settings related?
What is the role of the maui version? Programm behaviour?
What is the role of the workload version? Build and runtime?

Who decides the version of the maui depenencies version If no version is speified in the .csproj? Visual Studio?
- Microsoft.Maui.Dependencies
- Microsoft.Maui.Extensions

So to keep the tooling the same, should we define the maui versions in all projects?
`<MauiVersion>6.0.312</MauiVersion> `

What is the dependency between the visual studio and the maui version?
Can I use the latest maui version (e.g. "6.0.424)" with an older visual studio (e.g. 17.3.0 Preview 1.1)?

I observed that updating the visual studio does not update the the dotnet workload.
Do I need to manually set the version or does visual studio anyway use internally a different version?
Can I force in the cspro file to use a specific version or is this anyway done with the MauiVersion?

Is there something I have to take care for myself in respect to the version matching between maui version and workload?
```
dotnet workload list
Installierte Workload-IDs      Manifest Version      Installationsquelle
------------------------------------------------------------------------
runtimes-windows               6.0.7/6.0.300         VS 17.3.32721.290
maui-ios                       6.0.419/6.0.300       VS 17.3.32721.290
maui-android                   6.0.419/6.0.300       VS 17.3.32721.290
android                        32.0.440/6.0.300      VS 17.3.32721.290
maui-windows                   6.0.419/6.0.300       VS 17.3.32721.290
maccatalyst                    15.4.430/6.0.400      VS 17.3.32721.290
maui-maccatalyst               6.0.419/6.0.300       VS 17.3.32721.290
ios                            15.4.442/6.0.400      VS 17.3.32721.290
```

**Comment by epsmae** [comment]
In the mean time I found some answers and a workflow which works four our team.
The maui version 6.0.312 should use a dotnet sdk 6.0.3*. in order that it builds sucessful

We use now a global json and a .net maui version with that it works independent of the visual studio version.
global.json
```
{
  "sdk": {
    "version": "6.0.302",
    "rollForward": "latestPatch"
  }
}
```
for maui version:
`<MauiVersion>6.0.312</MauiVersion>`

The maui version can be fetched from https://www.nuget.org/packages/Microsoft.Maui.Dependencies as of now the release in this repo are usually delayed.

**Reply by dmccolloughOneGas** [reply]
@epsmae Be careful using MauiVersion, it's supposed to be not supported moving forward.
https://github.com/dotnet/maui/issues/4812

and then scroll down to the comment from Redth who is on the Maui team

**Reply by epsmae** [reply]
@dmccolloughOneGas thanks for your feedback currently we do not know another solution how to produce reproduceable builds on our ci/cd pipeline. 
@Redth you did not provide an alternative, what is the idea to handle different versions in future?
